compiler/src/dotty/tools/dotc/config/SourceVersion.scala

```diff
@@ -32,7 +34,7 @@ object SourceVersion extends Property.Key[SourceVersion]:
   def defaultSourceVersion = `3.6`
 
   /** language versions that may appear in a language import, are deprecated, but not removed from the standard library. */
-  val illegalSourceVersionNames = List("3.1-migration").map(_.toTermName)
+  val illegalSourceVersionNames = List("3.1-migration", "never").map(_.toTermName)
```

Comment:
I found that also this needs changing (the available choices to the `-source` flag:
https://github.com/scala/scala3/blob/56a384fd118c4a79218fd76e7ee2bb0cc12c83e3/compiler/src/dotty/tools/dotc/config/ScalaSettingsProperties.scala#L30-L31

Reply:
I have rolled that change into another PR (the named tuples one). 

Do you have time to do a full review by the end of the week? If we want to get this into 3.6 we need to be done by then.